data-structures/LinkedList.py

```diff
@@ -1,121 +1,110 @@
+    def __init__(self, value, next=None):
         self.value = value
-        self.next = next_node
+        self.next = next
 
 
 class LinkedList:
     def __init__(self):
         self.head = None
+        self.tail = None
         self.total = 0
 
     def add(self, value):
-        """ Add to the end of the list """
-        new_element = Node(value)
-        if not self.head:
-            self.head = new_element
-        else:
-            current = self.head
-            while current.next:  # while my current has a next Node
-                current = current.next
-            # add the element to the last node in the list
-            current.next = new_element
-
-        self.total += 1  # update total
+        """Add an element to the end of the list."""
+        new_item = Node(value)
+        if not self.head:  # If there is no element in the list
+            self.head = new_item
+            self.tail = new_item
+        else:              # If there are elements in the list
+            self.tail = new_item
+        self.total += 1
 
     def pop(self):
-        """ Remove the end of the list """
+        """Remove the last element of the list and return the value of the popped element."""
         if not self.head:
             return None
 
-        # if there is only one element
+        # If there is only one element
         if self.total == 1:
-            value = self.head.value
-            self.head = None  # removing the only element in the list
+            self.head = None # Removing the only element in the list
             self.total = 0
-            return value
 
-            # loop through and find last element
-        current = self.head
-        while current.next.next:  # find second to last element
-            current = current.next
+        # Loop through and find the last element
+        current_item = self.head
+        while current_item.next.next:          #Find second to last element
+            current_item = current_item.next
 
-        value = current.next.value
-        current.next = None
+        value = current_item.value
+        current_item.next = None
         self.total -= 1
         return value
 
-    def get_at_index(self, i):
-        """ return element at index i """
+    def get_value(self, i):
+        """Return element at the index i"""
         if i >= self.total or i < 0:
             raise IndexError
 
+        current_item = self.head
         current_index = 0
-        current = self.head
         while current_index != i:
-            current = current.next
+            current_item = current_item.next
             current_index += 1
+        return current_item.value
 
-        return current.value
-
-    def insert_at_index(self, i, value):
-        """ insert an element at a certain index i """
-
+    def insert(self, i, value):
+        """Insert an element at index i"""
         if i > self.total or i < 0:
             raise IndexError
 
-        if i == self.total:
```

Comment:
aren't you avoiding a performance boost by avoiding calling add to the end now that tail makes it faster?